frame/support/procedural/src/test_setup.rs

```diff
@@ -0,0 +1,279 @@
+}
+
+impl Parse for TestSetupAttr {
+	fn parse(input: ParseStream) -> syn::Result<Self> {
+		let mut attr = Self::default();
+		let mut lookahead = input.lookahead1();
+
+		macro_rules! parse_keyword {
+			($attr:ident, $input:ident, $lookahead:ident, $kw:ident, $camel_case:ident) => {
+				if $lookahead.peek(keyword::$kw) {
+					let _ = $input.parse::<keyword::$kw>();
+					$input.parse::<Token![=]>()?;
+					$attr.$camel_case = Some($input.parse::<syn::Type>()?);
+
+					if $input.peek(Token![,]) {
+						let _ = $input.parse::<Token![,]>();
+					}
+					$lookahead = $input.lookahead1();
+					continue;
+				}
+			}
+		}
+
+		while !input.is_empty() {
+			parse_keyword!(attr, input, lookahead, AccountData, account_data);
+			parse_keyword!(attr, input, lookahead, AccountId, account_id);
+			parse_keyword!(attr, input, lookahead, BaseCallFilter, base_call_filter);
+			parse_keyword!(attr, input, lookahead, BlockHashCount, block_hash_count);
+			parse_keyword!(attr, input, lookahead, BlockLength, block_length);
+			parse_keyword!(attr, input, lookahead, BlockNumber, block_number);
+			parse_keyword!(attr, input, lookahead, BlockWeights, block_weights);
+			parse_keyword!(attr, input, lookahead, Call, call);
+			parse_keyword!(attr, input, lookahead, DbWeight, db_weight);
+			parse_keyword!(attr, input, lookahead, Event, event);
+			parse_keyword!(attr, input, lookahead, Hash, hash);
+			parse_keyword!(attr, input, lookahead, Hashing, hash);
+			parse_keyword!(attr, input, lookahead, Header, header);
+			parse_keyword!(attr, input, lookahead, Index, index);
+			parse_keyword!(attr, input, lookahead, Lookup, lookup);
+			parse_keyword!(attr, input, lookahead, OnKilledAccount, on_killed_account);
+			parse_keyword!(attr, input, lookahead, OnNewAccount, on_new_account);
+			parse_keyword!(attr, input, lookahead, OnSetCode, on_set_code);
+			parse_keyword!(attr, input, lookahead, Origin, origin);
+			parse_keyword!(attr, input, lookahead, PalletInfo, pallet_info);
+			parse_keyword!(attr, input, lookahead, SystemWeightInfo, system_weight_info);
+			parse_keyword!(attr, input, lookahead, SS58Prefix, ss58_prefix);
+			parse_keyword!(attr, input, lookahead, Version, version);
+
+			// If we reach this point, none of the keywords were successfully parsed
+			return Err(lookahead.error())
+		}
+		Ok(attr)
+	}
+}
+
+pub fn setup_default_system_test_config(
+	attr: TokenStream,
+	item: TokenStream,
+) -> syn::Result<TokenStream> {
+	let ItemMacro { mac, .. } = syn::parse(item)?;
+	let attr: TestSetupAttr = if !attr.is_empty() { syn::parse(attr)? } else { Default::default() };
+
+	match mac.path.segments.last() {
+		Some(segment) if segment.ident == "construct_runtime" => {},
+		_ => {
+			let msg =
+				"Invalid test setup attribute macro: expected `construct_runtime!` macro call";
+			let span = mac.path.span();
+			return Err(syn::Error::new(span, msg))
+		},
+	}
+
+	let frame_system = generate_crate_access_2018("frame-system")?;
```

Comment:
I'm no big fan of requiring random crates, that are not the ones that re-export the macro.

We should at least improve the error message and to say why this required etc.

Reply:
unfortunately the macro `pallet` already rely on both frame-system and frame-support.

I was thinking the best would be to actually merge both crates. But I agree improving the error message would be great.